x/erc20/keeper/precompiles.go

```diff
@@ -42,11 +47,18 @@ func (k Keeper) InstantiateERC20Precompile(ctx sdk.Context, contractAddr common.
 	if !ok {
 		return nil, fmt.Errorf("token pair not found: %s", address)
 	}
+
+	if isNative {
+		return werc20.NewPrecompile(pair, k.bankKeeper, k.authzKeeper, *k.transferKeeper)
+	}
+
 	return erc20.NewPrecompile(pair, k.bankKeeper, k.authzKeeper, *k.transferKeeper)
 }
 
-// IsAvailableDynamicPrecompile returns true if the given precompile address is contained in the
-// EVM keeper's available dynamic precompiles precompiles params.
+// IsAvailableERC20Precompile returns true if the given precompile address
+// is contained in the params of the erc20 module.
+// Available precompiles are defined as the dynamic precompiles and the native
+// ones.
```

Comment:
Agree also with this, but I would not change it for the same reason of the previous reply. I would refactor all names + docstring in a single PR addressing the precompiles design

Reply:
I agree to do the other changes on a different repo but I think this doc comment we could merge on this PR because the doc comment itself was only added here, so why not directly use the more precise version?